### PR TITLE
Refactor core/cowdata.h

### PR DESCRIPTION
**Pull request by Cradmon**
- Remove excess casting
- Removed parameter `p_data` from `unref(void *p_data)`, as it was just passing `_ptr` as `p_data`.
- Renamed variables
- Updated comments

### CODE REVIEW

File: core/cowdata.h [hunk 1]
```diff
@@ -53,33 +53,30 @@ class CowData {
 private:
 	mutable T *_ptr;
 
-	// internal helpers
+	/* Internal Helpers */
```
Comment (LikeLakers2) [inline core/cowdata.h]: Just being a bit nit-picky here, but why the change to the comment style? It's a single-line comment, shouldn't it be `// comment`? Forgive me if you're seeing something I'm not.

Comment (Cradmon) [reply core/cowdata.h]: You're right that there is no real difference here, but I was just testing a comment style system that I mention in issue #21696. At this stage I'm just looking for feedback on this issue, and believe code reviewers are in a good position to objectively judge a system like this :)
Ultimately this pr isn't about the comment style and I can revert it back to a single line comment, if that is desired.

**Comment by akien-mga** [comment]
As mentioned in https://github.com/godotengine/godot/pull/22997#issuecomment-494148821, we decided in today's PR review meeting ([logs](http://godot.eska.me/irc-logs/meeting/2019-05-20.log)) not to merge those changes in their current form for the time being. Thanks for the contribution nevertheless!
